python_modules/dagster/dagster_tests/daemon_tests/conftest.py

```diff
@@ -41,21 +48,26 @@ def workspace_load_target(attribute=None):
 
 
 @pytest.fixture(name="workspace_context", scope="module")
-def workspace_fixture(instance_module_scoped):
+def workspace_fixture(instance_module_scoped) -> Generator[WorkspaceProcessContext, None, None]:
     with create_test_daemon_workspace_context(
         workspace_load_target=workspace_load_target(), instance=instance_module_scoped
     ) as workspace_context:
         yield workspace_context
 
 
 @pytest.fixture(name="external_repo", scope="module")
-def external_repo_fixture(workspace_context):
-    return next(
-        iter(workspace_context.create_request_context().get_workspace_snapshot().values())
-    ).repository_location.get_repository("the_repo")
+def external_repo_fixture(
+    workspace_context: WorkspaceProcessContext,
+) -> Generator[ExternalRepository, None, None]:
```

Comment:
Looks like the type you are casting to and the return type don't match?

Reply:
I'm returning an attribute of the thing I'm casting, not the thing I'm casting

Reply:
ah I see now, was confused by too-narrow diff window